File: Linked List/1. LinkedList/1. LinkedList Creation/LinkedList.java
/**
 * This class is the nodes of the SinglyLinked List.
 * They consist of a value and a pointer to the node after them.
 */
class Node {

    int value;  // Value of the node
    Node next;  // Point to the next node

    // Contructor
    Node ()
    {
        this.next = null;
    }
    Node(int value) {
        this.value = value;
        this.next = null;
    }
}

public class LinkedList
{
    private Node head;

    public LinkedList()
    {
        head = null;
    }

    public boolean isEmpty()
    {
        if (head == null)
            return true;
        else
            return false;
    }

    /*
     * Main method
     */
    public static void main(String args[])
    {
        LinkedList list = new LinkedList();
        System.out.println(list.isEmpty());
    }
}

File: Linked List/1. LinkedList/1. LinkedList Creation/LinkedList.cpp
#include <iostream>

using namespace std;

class Node{
    public:
        int data;               // Data to store (could be int,string,object etc)
        Node* nextElement;      // Pointer to next Element

        // Constructor to initialize nextElement of newly created node
        Node()
        {
            nextElement = nullptr;
        }
};

class LinkedList
{
    private:
        Node* head;
    public:
        LinkedList()
        {
            head = nullptr;
        }

        bool isEmpty()
        {
            if (head == nullptr)
                return true;
            else
            {
                return false;
            }    
        }
};

int main()
{
    LinkedList list; // LinkedList created
    cout << list.isEmpty() << endl;
}